api/types/authentication.go

```diff
@@ -314,62 +320,119 @@ func (c *AuthPreferenceV2) SetType(s string) {
 
 // GetSecondFactor returns the type of second factor.
 func (c *AuthPreferenceV2) GetSecondFactor() constants.SecondFactorType {
-	return c.Spec.SecondFactor
+	if c.Spec.SecondFactor != "" {
+		return c.Spec.SecondFactor
+	}
+
+	// If SecondFactor isn't set, try to convert the new SecondFactors field.
+	return legacySecondFactorFromSecondFactors(c.Spec.SecondFactors)
+}
+
+// legacySecondFactorFromSecondFactors returns a suitable legacy second factor for the given list of second factors.
+func legacySecondFactorFromSecondFactors(secondFactors []SecondFactorType) constants.SecondFactorType {
+	hasOTP := slices.Contains(secondFactors, SecondFactorType_SECOND_FACTOR_TYPE_OTP)
+	hasWebAuthn := slices.Contains(secondFactors, SecondFactorType_SECOND_FACTOR_TYPE_WEBAUTHN)
+
+	switch {
+	case hasOTP && hasWebAuthn:
+		return constants.SecondFactorOn
+	case hasWebAuthn:
+		return constants.SecondFactorWebauthn
+	case hasOTP:
+		return constants.SecondFactorOTP
+	default:
+		return constants.SecondFactorOff
+	}
 }
 
 // SetSecondFactor sets the type of second factor.
 func (c *AuthPreferenceV2) SetSecondFactor(s constants.SecondFactorType) {
 	c.Spec.SecondFactor = s
+
+	// SecondFactors takes precedence, so if SecondFactor is being set we should
+	// unsetSecondFactors here. SecondFactors will still be derived from SecondFactor
+	// configuration and set by CheckAndSetDefaults.
+	c.Spec.SecondFactors = nil
```

Comment:
if our goal is to migrate away from string, should we convert the second factor to SecondFactors?